tooling/nargo_cli/src/cli/mod.rs

```diff
@@ -83,22 +115,8 @@ pub(crate) fn start_cli() -> eyre::Result<()> {
     }
 
     // Search through parent directories to find package root if necessary.
-    match &command {
-        NargoCommand::Check(..)
-        | NargoCommand::Fmt(..)
-        | NargoCommand::Compile(..)
-        | NargoCommand::Execute(..)
-        | NargoCommand::Export(..)
-        | NargoCommand::Debug(..)
-        | NargoCommand::Test(..)
-        | NargoCommand::Info(..) => {
-            config.program_dir = find_package_root(&config.program_dir)?;
-        }
-        NargoCommand::New(..)
-        | NargoCommand::Init(..)
-        | NargoCommand::Lsp(..)
-        | NargoCommand::Dap(..)
-        | NargoCommand::GenerateCompletionScript(..) => (),
+    if let Some(workspace) = is_workspace_rooted(&command) {
+        config.program_dir = nargo_toml::find_root(&config.program_dir, workspace)?;
```

Comment:
This is a nice refactor but this causes an unwanted change in behaviour, I think. We want to move to the root of any packages we're in for these commands so that we can run `nargo execute` anywhere within a noir package rather than just at the root. The issue is it seems we only do so if we've got the `--workspace` flag set now.

Reply:
Ohhh, I'm being slow. The `Option` handles this.

Reply:
A enum wouldn't be the worst thing in the world tbh, using an `Option` for this puts of a little bit of a hurdle for people reading this.

```noir 
fn get_command_location(&Command) -> CommandLocation

enum CommandLocation {
   Workspace,
   Package,
   Any
}
````

Reply:
I started adding something like that, but then it becomes so so noisy to invoke `find_root`. Renamed to `manifest_scope` now with an explanation. But if you think an enum is better I'll give it another go.

Reply:
It's an enum now 👍